probe.go

```diff
@@ -257,7 +257,12 @@ func (p *Probe) GetLastError() error {
 
 // ID returns the system-wide unique ID for this program
 func (p *Probe) ID() uint32 {
-	return uint32(p.systemWideID)
+	p.stateLock.RLock()
+	defer p.stateLock.RUnlock()
+	if p.state >= initialized {
```

Comment:
Is this `if` needed? The value only gets set during initialization, so it will already either be `0` or valid.

Reply:
You're right, it's not needed. I removed it in my last commit.